### PR TITLE
Feature/34 summaries

### LINKED ISSUE #34: BaseSummaryModule
**What is your feature request related to?**
* Problem related

**Please describe the problem**

Sometimes the functionality of modules is to calculate statistics of different datasets. E.g., to compare the statistics of two datasets. Currently, for solving this it is necessary that the module returns a Dataset which is not  related to the functionality of the module. 

**Describe the solution you'd like**
A new summaryBase class, from which such modules can inherit. The stepFactory can now check, if such a module is used as input for an other, and raise an exception in that case. 

**What type of Feature Request**
* Core enhancement

### PR DESCRIPTION
**Pull request by benHeid**
## Description
Add a new kind of modules, the summary modules. These summary Modules perform a summary operation e.g. evaluating the result  of some outputs and provide a summary as markdown text. Consequently, their output must not be used by other modules. Therefore, the SummaryStep and the corresponding information is created.

## Related Issues
closes #34 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos)
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Did you update the CHANGELOG

## PR review
Anyone in the community is free to review the PR once the tests have passed.

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones to the PR so it can be classified

### CODE REVIEW

**Comment by benHeid** [comment]
## TODOs

* [ ] Is a markdown summary suitable? How about images?
* [ ] The documentation has to make a clear differentiation to callbacks.

File: pywatts/modules/rolling_rmse.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from pywatts.core.base import BaseTransformer
+from pywatts.core.exceptions.input_not_available import InputNotAvailable
+from pywatts.utils._xarray_time_series_utils import _get_time_indeces
+
+
+class RollingRMSE(BaseTransformer):
```
Comment (olineumann) [inline pywatts/modules/rolling_rmse.py:12]: Maybe we could restructure the pyWATTS library to pyWATTS.metrics, pyWATTS.core, pyWATTS.models, ... to give the user some structure.

**Review by olineumann — APPROVED** [review]
Nice work! I found no bug, issue, complain, ... :+1:

File: pywatts/modules/rolling_rmse.py (hunk 1 above)
Comment (benHeid) [reply pywatts/modules/rolling_rmse.py:12]: Good idea, I open an issue

**Pull request merged**